valuable/src/structable.rs

```diff
@@ -68,32 +155,134 @@ impl fmt::Debug for dyn Structable + '_ {
 }
 
 impl<'a> StructDef<'a> {
+    /// Create a new `StructDef::Static` instance.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use valuable::{StructDef, Fields};
+    ///
+    /// let def = StructDef::new_static("Foo", Fields::Unnamed);
+    /// ```
     pub const fn new_static(name: &'static str, fields: Fields<'static>) -> StructDef<'a> {
         StructDef::Static { name, fields }
     }
 
+    /// Create a new `StructDef::Dyanmic` instance.
+    ///
```

Comment:
and maybe something like
```suggestion
    ///
    /// This is used when the struct's fields may vary at runtime.
```